snap/plugins/x-php.py

```diff
@@ -99,7 +99,75 @@ def clean_pull(self):
 
+    def __fix_libpath_getDist(self):
+        # Because my alternative to the missing "endif" is goto....
+        # Get the current distro.
+        distCheck_raw = subprocess.check_output("if [ \"`lsb_release -is`\" = \'Ubuntu\' ] ; then echo 1; else echo 0; fi;", shell=True)
+        distCheck = int(distCheck_raw.decode('utf-8'))
+        return distCheck
+
+    def __fix_libpath_getPath(self, distCheck):
+        # Because my alternative to the missing "endif" is goto....
+        # Get the current library path. (If needed)
+        if distCheck == 1:
+            # Ubuntu. We need to set the lib directory to the correct tupple for the system arch.
+            libPath_raw = subprocess.check_output("gcc -dumpmachine", shell=True)
+            libPath = ('lib/' + libPath_raw.decode('utf-8')).rstrip('\r\n')
+            return libPath
+        else:
+            # The default is to not modifiy anything.
+            libPath = ''
+            return libPath
+
+    def __fix_libpath_fixFlag(self, flag, libPath, count):
+        # Check for the correct flag and fix it if found.
+        if "--with-libdir=" in flag:
+            # Got a match fix the flag.
+            fixed_flag = '--with-libdir=' + libPath
+            logger.info('Fixing libdir flag at configflag index ' + str(count) + ' to: \'' + fixed_flag + '\'')
+            del self.options.configflags[count]
+            self.options.configflags.insert(count, fixed_flag)
+
+    def __fix_libpath_fixFlag_loop(self, libPath):
+        # Create a counter.
+        count = 0
+
+        # Check for pre-existing flags.
+        for flag in self.options.configflags:
+            self.__fix_libpath_fixFlag(flag, libPath, count)
+
+            # Increment count.
+            count = count + 1
+
+    def __fix_library_path(self):
+        # Set up the vars.
+        libPath = ''
+        distCheck = 0
+
+        # Get the current library path.
+        distCheck = self.__fix_libpath_getDist()
+        libPath = self.__fix_libpath_getPath(distCheck)
+
+        # Check to see if we need to continue. (We abort if distCheck is zero.)
+        if not distCheck == 0:
+            logger.info('Modifications to --with-libdir flag may be required.')
+            if not libPath == "":
+                if self.options.configflags:
+                    logger.info('Checking configflags....')
+
+                    # Call loop function.
+                    self.__fix_libpath_fixFlag_loop(libPath)
+                else: # self.options.configflags
+                    logger.info('No configflags defined???')
+            else: #not libPath == ""
+                logger.info('No libPath defined, cannot fix --with-libdir flag without it. The build may fail.')
+        else: # not distCheck == 0
+            logger.info('No modifications to --with-libdir flag required.')
+
     def build(self):
+        # Check for a broken libpath.
+        self.__fix_library_path()
```

Comment:
Please help me understand this. This takes the `--with-libdir=` used [here](https://github.com/nextcloud/nextcloud-snap/pull/411/files#diff-5fde7a6d86053f0e1d88c0a2a238941fR167), and turns it into `--with-libdir=/lib/x86_64-linux-gnu` (on amd64, of course)? Why is this required? And it only seems to do this on Ubuntu-- why is it not required elsewhere?

Reply:
I believe this is to get around a quirk with Ubuntu, which is mentioned in this comment on the issue: https://github.com/nextcloud/nextcloud-snap/issues/117#issuecomment-256225142 .  I copied the code that @phibbs7 created here (https://github.com/phibbs7/nextcloud-snap/commit/e85418393f407840c5a913990208ae2c91dac0c6) and applied it to the current develop branch.  His fork was out of date and would not build the snap without several updates to the snapcraft.yml

Reply:
Ah yes, I recall that conversation now.

Snapcraft v2.36 has a way of dealing with this much better, by making `SNAPCRAFT_ARCH_TRIPLET` available for use in the YAML. Only v2.35 is available as a deb right now, though, but we can use that approach as inspiration for a much simpler way to make this work. Then once 2.36 is out it'll be easy to remove this small hack and start using that instead.